docs(defi): update ckBTC docs (#6096)
dfinity/portal: docs/defi/chain-key-tokens/ckbtc/using-ckbtc-in-dapps.mdx

Author: I kind of agree, but I would redo the whole docs in that regard if we decide to go that route. for now I would leave it as is 😅

This means that integration with the ledger involves calling the standard endpoints for actions such as:

- `icrc1_balance_of` to fetch balances,  
- `icrc1_transfer` to transfer tokens, and
- `icrc2_approve` / `icrc2_transfer_from` for delegated transfers.

There are two primary ways to interact with the ckBTC ledger:

### Client calls (via agent)
A client application (e.g. web application) can use an [agent](/docs/building-apps/interact-with-canisters/agents/overview) to interact directly with the ckBTC ledger canister.

An **agent** is the API client layer used to call ICP canisters. For example, the [ICP JavaScript agent](/docs/building-apps/interact-with-canisters/agents/javascript-agent) can create an actor from the ledger’s [Candid interface](/docs/building-apps/interact-with-canisters/candid/candid-concepts) to invoke its endpoints.

The following libraries simplify creating actors for the ckBTC ledger:

- Rust: [icrc-ledger-agent](https://crates.io/crates/icrc-ledger-agent)
- TypeScript: [@dfinity/ledger-icrc](https://www.npmjs.com/package/@dfinity/ledger-icrc)

> In this model, all transactions are executed from the user’s own [account](/docs/defi/token-standards/#accounts-principals-and-subaccounts) (typically the principal without a subaccount).

Reviewer: ```suggestion
In this model, all transactions are signed by the user’s [account](/docs/defi/token-standards/#accounts-principals-and-subaccounts).
```

### Inter-canister calls
A canister can also interact with the ckBTC ledger.

- In some cases, the canister may use ckBTC for its own purposes, such as making payments to other canisters, managing a treasury, or executing automated transfers.
- In other cases, the canister may act on behalf of users, which requires managing user funds. This can be achieved by creating **subaccounts per user** or by **pooling all funds in a single account** and maintaining an internal balance table.

>   - It is recommended to use the [ICRC-2](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-2/README.md) approval flow (`icrc2_approve` / `icrc2_transfer_from`) to move funds from a user’s wallet to the backend canister.

Reviewer: i'd personally not say that this is recommended. many roads lead to rome :)

Libraries are available to simplify inter-canister calls:

> - Rust: [icrc-ledger-client-cdk](https://crates.io/crates/icrc-ledger-client-cdk)

Reviewer: i'd mention this as well

https://crates.io/crates/icrc-ledger-client
https://crates.io/crates/icrc-ledger-types

Author: see above. my understanding was that I need to use `icrc-ledger-client-cdk` (which relies on `icrc-ledger-client` that uses `icrc-ledger-types`).

can you elaborate on this shortly?

I don't want to confuse people too much with mentioning different libraries which aren't well documented.

- Motoko: [icrc2-types](https://mops.one/icrc2-types) (includes ICRC-1 types)

Example code:
- Motoko: [transfer](https://github.com/dfinity/examples/tree/master/motoko/token_transfer), [transfer_from](https://github.com/dfinity/examples/tree/master/motoko/token_transfer_from)

> - Rust: [transfer](https://github.com/dfinity/examples/tree/master/rust/token_transfer), [transfer_from](https://github.com/dfinity/examples/tree/master/rust/token_transfer_from)

Reviewer: what about this example?
https://github.com/dfinity/examples/tree/master/rust/token_transfer_from

Author: this is actually there 🤔

## Bitcoin ↔ ckBTC conversion

Beyond ledger interactions, a dapp may also support minting and burning ckBTC by calling the [ckBTC minter canister](https://dashboard.internetcomputer.org/canister/mqygn-kiaaa-aaaar-qaadq-cai).

This can be done in two ways:

- **Client calls (via agent)**
  - Any client can call the ckBTC minter canister directly.
  - TypeScript library: [@dfinity/ckbtc](https://www.npmjs.com/package/@dfinity/ckbtc)

- **Inter-canister calls**
  - A backend canister can call the ckBTC minter canister as part of its logic, either for its own use or on behalf of users.

Learn more about implementing this conversion flow in the [ckBTC best practices](/docs/defi/defi-best-practices).

### Why integrate conversion into a dapp
- Enables deposits from Bitcoin → ckBTC and withdrawals from ckBTC → Bitcoin without leaving the application.
- Reduces friction and improves overall user experience.

**Note:** Applications that integrate with [OISY](https://oisy.com) may rely on OISY to handle Bitcoin ↔ ckBTC conversion on behalf of their users, instead of implementing the conversion themselves.

> ## Examples applications for ckBTC

Reviewer: ```suggestion
## Example applications for ckBTC
```

> - [Bitcoin point of sale app](/docs/references/samples/motoko/ic-pos/).

Reviewer: if there is only one example we can show here, should we remove this section? looks a bit empty otherwise

Author: I might find another good one. let me see.
